sympy/printing/latex.py

```diff
@@ -1813,7 +1813,50 @@ def _print_PolynomialRingBase(self, expr):
 
     def _print_Poly(self, poly):
         cls = poly.__class__.__name__
-        expr = self._print(poly.as_expr())
+        terms, gens = [], [s for s in poly.gens]
+        for monom, coeff in poly.terms():
+            s_monom = None
+            for i, exp in enumerate(monom):
+                if exp > 0:
+                    if exp == 1:
+                        s_monom = self._print(gens[i])
```

Comment:
This overwrites the previous value of s_monom every time, so only the last of the generators (the variables of the polynomial) survives. For example, 
```
>>> latex(Poly(x**2*y**2 + x*y, (x, y)))
'\\operatorname{Poly}{\\left( y^{2} + y, x, y, domain=\\mathbb{Z} \\right)}'
```
You probably want `s_monom += ...` here. Compare to what str printer does (it prints correctly).

For testing purposes, try something like 
```
latex(Poly(a*x**3 + x**2*y - x*y - c*y**3 - b*x*y**2 + y - a*x + b, (x, y)))
```